### PR TITLE
Add deleteDraftProject GQL mutation

### LINKED ISSUE #795: Create proper deleteDraftProject GQL mutation
Related to #794, but a slightly different concern than #794 so I'm creating this issue to track the GraphQL part specifically.

### PR DESCRIPTION
**Pull request by rmunn**
Fixes #795.

### CODE REVIEW

File: backend/LexBoxApi/GraphQL/ProjectMutations.cs
Comment (myieye) [inline backend/LexBoxApi/GraphQL/ProjectMutations.cs:360]: This will always return an empty list right?

**Review by myieye — CHANGES_REQUESTED** [review]
1 comment as well as this one 🙂: I think it makes sense to remove the draft-project logic from our other delete mutation.

File: backend/LexBoxApi/GraphQL/ProjectMutations.cs
Comment (hahn-kev) [reply backend/LexBoxApi/GraphQL/ProjectMutations.cs:360]: Yeah, I'd say we query the draft out so we can return it here, as an object.

File: backend/LexBoxApi/GraphQL/ProjectMutations.cs
Comment (rmunn) [reply backend/LexBoxApi/GraphQL/ProjectMutations.cs]: Done.

File: backend/LexBoxApi/GraphQL/ProjectMutations.cs
Comment (rmunn) [reply backend/LexBoxApi/GraphQL/ProjectMutations.cs]: Done.

**Comment by rmunn** [comment]
> 1 comment as well as this one 🙂: I think it makes sense to remove the draft-project logic from our other delete mutation.

Also done.

File: frontend/src/routes/(authenticated)/admin/AdminProjects.svelte
Comment (myieye) [inline frontend/src/routes/(authenticated)/admin/AdminProjects.svelte:64]: Sorry, for a late comment, but how about we move the `if (project.isDraft)` logic here and rename this method to `deleteProjectOrDraft` instead of having 2 essentially duplicate methods. E.g.
const { error } = await (project.isDraft ? _deleteDraftProject(project.id) : _deleteProject(project.id));

Comment (rmunn) [reply frontend/src/routes/(authenticated)/admin/AdminProjects.svelte:64]: That's how I would prefer to write it, as I prefer functional-style code. But I've had functional-style code rejected in previous code reviews, so I'm never quite certain how much functional style is "too much". I'll switch it to this style tomorrow.

Comment (hahn-kev) [reply frontend/src/routes/(authenticated)/admin/AdminProjects.svelte:64]: I'm fine with this also. I would say too far would be passing a function as a parameter which actually executes the delete.

**Review by hahn-kev — APPROVED** [review]

**Comment by rmunn** [comment]
@myieye - Now that Kevin has approved this (modulo the change that I just made), want to approve it too so GitHub will unlock the merge button?

**Review by myieye — APPROVED** [review]

**Pull request merged**
